scripts/launch-s0-instance.sh

```diff
@@ -3,81 +3,233 @@
 me=$(basename $0)
 dir=$(readlink -f $(dirname $0))
 elastio_aws_id=537513441174
-security_group="elastio-s0-server"
+default_security_group="elastio-s0-server"
 default_instance_type="m5d.2xlarge"
 default_instance_name="$(whoami)-s0-server"
+default_block_size=fixed
+bootstrap=/tmp/s0-bootstrap.yml
+db_mount=/mnt/elastio
+unit_file=/etc/systemd/user/s0.service
+service=s0
+
+create_s0_bootstrap()
+{
+    rm -f $bootstrap
+    cat << EOF > $bootstrap
+#cloud-config
+# Bootstrap a 's0' server on Amazon Linux 2.
+# This script is intended to run as part of the cloud init
+
+write_files:
+  - path: /home/ec2-user/.aws/config
+    owner: ec2-user:ec2-user
+    content: |
+      [default]
+      output = json
+
+  - path: $UNIT_FILE
+    owner: ec2-user:ec2-user
+    permissions: '0664'
+    content: |
+      [Unit]
+      Description="Elastio Scale-to-Zero service"
+      After=network.target
+
+      [Service]
+      User=ec2-user
+      StartLimitInterval=5
+      StartLimitBurst=10
+      TimeoutStopSec=5min
+      ExecStart=/bin/bash -c '/usr/bin/s0 --log-output compact vault serve s3 --bucket $S0_BUCKET --key-id $S0_KMS $BLOCK_SIZE --server-address 0.0.0.0 --db-path $DB_MOUNT 2>&1 | sed -r "s/\x1B\[([0-9]{1,3}(;[0-9]{1,2})?)?[mGK]//g" >> /var/log/s0-service.log'
+      Restart=always
+      RestartSec=5
+
+      [Install]
+      WantedBy=multi-user.target
+
+runcmd:
+  # Format and mount nvme disk
+  - |
+    mkdir -p $DB_MOUNT
+    if [[ -b /dev/nvme1n1 && -b /dev/nvme2n1 ]]; then
+      # Two disks; make an array
+      mdadm --create --verbose /dev/md0 --level=0 --raid-devices=2 /dev/nvme1n1 /dev/nvme2n1
+      mkfs.xfs -f /dev/md0
+      mount /dev/md0 $DB_MOUNT
+    elif [ -b /dev/nvme1n1 ]; then
+      # One disk; no need to involve mdadm
+      parted -a optimal --script /dev/nvme1n1 mklabel gpt
+      parted -a optimal --script /dev/nvme1n1 mkpart primary 0% 100%
+      mkfs.xfs -f /dev/nvme1n1p1
+      mount /dev/nvme1n1p1 $DB_MOUNT
+    fi
+    chown -R ec2-user:ec2-user $DB_MOUNT
+
+  # Configure AWS region
+  - |
+    echo "region = \$(curl -s http://169.254.169.254/latest/dynamic/instance-identity/document | grep -w region | cut -d: -f2 | tr -d '", ')" >> /home/ec2-user/.aws/config
+    cp -r /home/ec2-user/.aws /root/
+
+  # Create s0 vault
+  - |
+    s0 vault create s3 --bucket $S0_BUCKET --key-id $S0_KMS 2>&1 | sed -r "s/\x1B\[([0-9]{1,3}(;[0-9]{1,2})?)?[mGK]//g" > /var/log/s0-vault-create.log
+
+  # Prepare and start s0 service
+  - |
+    touch /var/log/s0-service.log
+    chown ec2-user:ec2-user /var/log/s0-service.log
+    systemctl daemon-reload
+    systemctl enable $UNIT_FILE
+    systemctl start $SERVICE
+
+output:
+  all: '| tee -a /var/log/cloud-init-output.log'
```

Comment:
Why is this needed?  I always `tail -f /var/log/clout-init-output.log` to monitor the progress of cloud init tasks, and I never out this `output` element in my cloud init scripts.

Reply:
I found this in the AWS doc https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/user-data.html and used for debugging to resolve issues of the bootstrap scripts. However I found very similar log file inside of VM `/var/log/cloud-init.log`. Probably you are referring to it. But `/var/log/cloud-init-output.log` doesn't exist by default.
If you like, I can remove it because debugging is done.

Reply:
Strange.  I've always used `cloud-init-output.log` and I never used a directive like this to create it.  I wonder what the difference is.

Anyway it's not hurting anything.  That's just an FYI.